app/channel.go

```diff
@@ -1112,7 +1114,9 @@ func (a *App) postAddToTeamMessage(user *model.User, addedUser *model.User, chan
 		UserId:    user.Id,
 		RootId:    postRootId,
 		Props: model.StringInterface{
+			"userId":        user.Id,
```

Comment:
Is this userId needed? why not use the Post.UserId?

Reply:
Actually, that's my proposal. On [client](https://github.com/philippe-granet/mattermost-webapp/blob/646027624ae98dbd8e0489e9f91295645800b2ec/components/post_markdown/system_message_helpers.jsx#L50), it should be like:
```
function renderAddToChannelMessage(post, options) {
    const username = renderUsernameForUserId(post.props.user_id, options);
    const addedUsername = renderUsernameForUserId(post.props.addedUserId, options);
```

I have a WIP for consolidated system message that will change above to something like:
```
function renderAddToChannelMessage(postProps, options) {
    const username = renderUsernameForUserId(postProps.user_id, options);
    const addedUsername = renderUsernameForUserId(postProps.addedUserId, options);
```

Reply:
Looks like some kind of duplication, but probably you have more context here. I don't know exactly why we need to duplicate it. I can imagine a similar solution with:

```javascript
function renderAddToChannelMessage(userId, addedUserId, options) {
    const username = renderUsernameForUserId(userId, options);
    const addedUsername = renderUsernameForUserId(addedUserId, options);
    ...
}
```

Reply:
Currently, we are just passing `post` (and `options`) to [systemMessageRenderers](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_markdown/system_message_helpers.jsx#L315). With that, it's totally fine to use `post.user_id` now but it would be better to have `userId` available in `post.props` so that we could just pass it to `systemMessageRenderers` instead of the entire `post` in the future (which is the way I look when dealing with consolidated system messages).